### ISSUE TITLE
Site Content Disappearing until Save

### ISSUE THREAD

**Issue by bryantAXS** [spicywebau/craft-neo#227]
#### Description
After an update to 3.2 I'm seeing issues where content on the site gets deleted.

#### Steps to reproduce
1. Open entry fresh of a 3.2
2. Click Share
3. Content Disappears
4. Need to Resave content to appear on site.

Video: https://cl.ly/642e0f3d58ef

#### Other information
- Neo version: 2.3.5
- Craft version: 3.2
- Is the affected Craft installation multi-site: Yes
- (if multi-site) Are the affected Neo fields set to manage blocks on a per-site basis: Yes

**Comment by bryantAXS** [comment]
Also, I attempted to use the Craft CLI to resave all entries, but entries that I hadn't opened within the CP and saved at least once we're throwing a particular error.

https://cl.ly/6eac6a87ad64

Once I opened up those entries that did error, saved them, then reran the CLI command, it worked as intended.

**Comment by eliseferguson** [comment]
I've noticed a similar issue.  Pre-craft 3.2 I have an entry with several blocks from a Neo field.  Upgrade to 3.2 and nothing appears on the entry both front end and admin side.  I have been able to revert to a previous version of the page and get my neo blocks back.

**Comment by brandonkelly** [comment]
I’ve been able to reproduce this locally with @eliseferguson’s install. The data loss is happening during the “Updating entry drafts and revisions” background job that kicks off automatically after updating to Craft 3.2. It only affects entries that have old revisions which don’t have any Neo field data. In those cases, when Craft goes to create the old revision, Neo ends up reassigning the current revision’s blocks to that old revision.

I’ve just submitted a [pull request](https://github.com/spicywebau/craft-neo/pull/235) that fixes this, but unfortunately I don’t think there is a way to retroactively fix entries that have already lost their content. So I’m suggesting in the PR that Spicy releases the fix as Neo 2.3.5.1 for people running Craft 3.1, so they can get the fix installed at or before installing Craft 3.2, before that background job kicks off.

**Comment by bstein-clever** [comment]
We just ran into this issue on a Craft version upgrade yesterday. Several of our blocks went missing, but the previous version was extremely old, so restoring from that wasn't a good update.
I, too, would love a way to protect against this before we re-upgrade.

I also recall running into https://github.com/spicywebau/craft-neo/issues/174 the last time we did a big version bump. I'd love it there we could come up with some way to catch and fix them before they get reported. :(
